charts/camunda-platform/test/integration/login.go

```diff
@@ -0,0 +1,139 @@
+	//		method="post">
+	//
+	//
+	defer response.Body.Close()
+	body := response.Body
+	b, err := io.ReadAll(body)
+
+	regexCompiled := regexp.MustCompile("(action=\")(.*)(\"[\\s\\w]+=\")")
+	match := regexCompiled.FindStringSubmatch(string(b))
+	if len(match) < 3 {
+		return "", errors.New(fmt.Sprintf("Expected to extract session url from response %s", string(b)))
+	}
+	sessionUrl := match[2]
+
+	// the url is encoded in the html document, which means we need to replace some characters
+	return strings.Replace(sessionUrl, "&amp;", "&", -1), nil
+}
+
+func (s *integrationTest) extractJWTTokenFromCookieJar(jar *cookiejar.Jar) (string, error) {
+	cookies := jar.Cookies(&url.URL{Scheme: "http", Host: "localhost"})
+	identityJWT := "IDENTITY_JWT"
+	for _, cookie := range cookies {
+		if cookie.Name == identityJWT {
+			return cookie.Value, nil
+		}
+	}
+	return "", errors.New("no JWT token found in cookie jar")
+}
+
+func (s *integrationTest) doJWTBasedLogin(err error, jar *cookiejar.Jar, identityEndpoint string, httpClient http.Client) error {
+	// The previous log in request caused to store a token in our cookie jar.
+	// In order to verify whether this token is valid and works with identity we have to extract the token and set
+	// the cookie value (JWT token) as authentication header.
+	jwtToken, err := s.extractJWTTokenFromCookieJar(jar)
+	if err != nil {
+		return err
+	}
+	s.T().Logf("Extracted following JWT token from cookie jar '%s'.", jwtToken)
+
+	verificationUrl := "http://" + identityEndpoint + "/api/clients"
+	getRequest, err := http.NewRequest("GET", verificationUrl, nil)
+	if err != nil {
+		return err
+	}
+	getRequest.Header.Set("Authentication", "Bearer "+jwtToken)
+
+	// verify the token with the get request
+	getResponse, err := httpClient.Do(getRequest)
+	if err != nil {
+		return err
+	}
+
+	if getResponse.StatusCode != 200 {
+		return errors.New(fmt.Sprintf("On validating JWT token expected an 200 status code, but got %d", getResponse.StatusCode))
+	}
+	return nil
+}
+
+func (s *integrationTest) doLogin(service string, localPort int, containerPort int) (string, http.Client, func(), error) {
+	// In order to login to the service we need to port-forward to Keycloak.
+	// The service will redirect (forward) requests to Keycloak to enable the login
+
+	// create keycloak port-forward
+	keycloakServiceName := s.resolveKeycloakServiceName()
+	_, closeKeycloakPortForward := s.createPortForwardedHttpClientWithPort(keycloakServiceName, 18080)
+
+	// create service port-forward
+	serviceName := fmt.Sprintf("%s-%s", s.release, service)
+	endpoint, closeFn := s.createPortForwardedHttpClientWithPortAndContainerPort(serviceName, localPort, containerPort)
+
+	coupledCloseFn := func() { closeFn(); closeKeycloakPortForward() }
+	httpClient, _, err := s.createHttpClientWithJar()
+	if err != nil {
+		return "", http.Client{}, coupledCloseFn, err
+	}
+
+	err = s.doSessionBasedLogin("http://"+endpoint+"/api/login", httpClient)
```

Comment:
:question: Do we always call `doSessionBasedLogin` with this? Then we could just push `"http://"+endpoint+"/api/login"` into that, and just pass the endpoint here.

Reply:
Unfortunately, Identity uses a different endpoint https://github.com/camunda/camunda-platform-helm/blob/main/charts/camunda-platform/test/integration/integration_test.go#L274

Reply:
Yeah, I thought I might've missed one :+1: